enhancements/cluster-logging/content-filter.md

```diff
@@ -0,0 +1,199 @@
+
+A new `filters` section in the `ClusterLogForwarder` allows named filters to be defined.
+This proposal defines two types of filter "prune" and "drop".
+
+#### Prune filters
+
+A "prune" filter removes fields from each record passing through the filter.
+
+##### API
+
+``` yaml
+spec:
+  filters:
+  - name: ""        # User selects filter name
+    type: prune
+    prune:
+      in: []      # Array of field paths, remove fields in the array.
+      notIn: []   # Array of field paths, remove all fields that are NOT in the array.
+```
+
+##### Examples
+
+``` yaml
+  spec:
+    filters:
+    - name: foo
+      type: prune
+      prune:
+		notIn: [.message, .kubernetes]   # Keep only the message and kubernetes fields.
+        in: [.kubernetes.flat_labels]    # Prune the kubernetes.flat_labels sub-field.
+    pipelines:
+    - name: bar
+		filterRefs: ["foo"]
+```
+
+#### Drop filters
+
+##### API
+
+A drop filter applies a sequence of tests to a log record and drops the record if any test passes.
+Each test contains a sequence of conditions, all conditions must be true for the test to pass.
+
+``` yaml
+spec:
+  filters:
+  - name:             # Provided by the user
+    type: drop
+    drop:
+      - test:
+        - field:      # JSON path to the field
+		  # Requires exactly one of the following conditions.
+          matches:    # regular expression match
+          notMatches: # regular expression does not match
+```
+
+Note:
+- If _all_ conditions in a test are true, the test passes.
+- If _any_ test in the drop filter passes, the record is dropped.
+- If there is an error evaluating a condition (e.g. a missing field), that condition evaluates to false.
+  Evaluation continues as normal.
+
+The drop filter is equivalent to a boolean OR of AND clauses. Any boolean expression can be reduced to this form.
+
+##### Example
+
+``` yaml
+filters:
+  - name: important
+    type: drop
+    drop:
+      - tests:
+		- field: .kubernetes.namespace_name
+		  notMatches: "very-important"  # Keep everything from this namespace.
+		- field: .level # Keep important levels
+		  matches: "warning|error|critical"
+```
+
+### Implementation Details
+
+### JSON path
+
+Need to document the rules for field paths, these are a subset of the JSON path spec.
+In fact we will use the same subset as Vector does, but we should describe the rules explicitly
+and not refer to vector docs, so we don't create assumptions in the API about the underlying collector.
+
+### Metrics
+
+Vector provides metrics for records in/out of each vector node.
+Records dropped by filters can be computed from this.
+We should provide _brief_ hints and links to these metrics, but not duplicate vector docs.
+
+### Risks and Mitigations
+
+- Performance impact, need to benchmark slowdown due to rule evaluation, but balance against reduction in data volume. 
+- No new security issues.
+- Extensions to ClusterLogForwarder will be reviewed as part of normal code review.
+
+### Drawbacks
```

Comment:
Regexp syntax can be depended on collector implementation, e.g: Ruby is case-sensitive, Rust is case-insensitive by default and etc.